File: design_patterns/c++/src/create/abstract_factory/BombedMazeFactory.cpp
//
// Created by Albert on 2020/12/4.
//

#include "../../../headers/create/abstract_factory/BombedMazeFactory.h"
#include "../../../headers/Wall.h"
#include "../../../headers/bomb/bomb_room.h"
#include "../../../headers/bomb/BombedWall.h"

Wall* BombedMazeFactory::makeWall() const {
  return new BombedWall;
}

Room* BombedMazeFactory::makeRoom(int num) const {
  return new RoomWithABomb(num, true);
}

File: design_patterns/java/src/create/abstract_factory/BombedMazeFactory.java
package create.abstract_factory;

import base.Room;
import base.Wall;
import base.bomb.BombedWall;
import base.bomb.RoomWithABomb;

public class BombedMazeFactory extends MazeFactory {

    @Override
    public Wall makeWall() {
        return new BombedWall();
    }

    @Override
    public Room makeRoom(int num) {
        return new RoomWithABomb(num, true);
    }

}
